pandas/core/frame.py

```diff
@@ -8630,7 +8632,9 @@ def _get_data(axis_matters):
                 raise NotImplementedError(msg)
             return data
 
-        if numeric_only is not None and axis in [0, 1]:
+        if (numeric_only is not None and axis in [0, 1]) or (
```

Comment:
i think this is worth a comment about what is being selected here as this is a complicated expression.

Reply:
will do

Reply:
updated+green